ctru-rs/src/lib.rs

```diff
@@ -7,40 +7,28 @@
 #![feature(nonnull_slice_from_raw_parts)]
 #![test_runner(test_runner::run)]
 
-extern "C" fn services_deinit() {
-    unsafe {
-        ctru_sys::psExit();
-    }
-}
+// These functions are imported to assure ´cargo´ we need to link the crates.
+// These don't need to run to link the functions properly!
+#[allow(unused_imports)]
+use linker_fix_3ds::init as link_init;
+#[allow(unused_imports)]
+use pthread_3ds::init as pthread_init;
```

Comment:
could these perhaps be 
```rs
extern crate linker_fix_3ds;
extern crate pthread_3ds;
```

I guess I'm not sure exactly what criteria `cargo` uses to decide to discard an entire crate during linking...

I wonder if [this Cargo feature](https://rust-lang.github.io/rfcs/3028-cargo-binary-dependencies.html) would ideally allow us to depend on them as `artifact = "staticlib"` so they participate in linking, but avoid any extra stuff in the code?

Reply:
I didn’t try just importing the crate, I forgot about that possibility. Though yes, cargo seems to not compile/pass to the linker crates which have zero references. I imported those functions just because they existed already.